### ISSUE TITLE
Problem in reproducing FEP results

### ISSUE THREAD

**Issue by zfwangDP** [BFeng14/ActFound#9]
I am trying to reproduce the FEP results with experimental bioactivity as fine-tuning data,  but the results i get does not matches the provided ones (in test_results_all.tar.gz). The average r2 is 0.449,0.550,0.604 and 0.587 for sup_num 0.2,0.4,0.6,0.8 in my experiment.  Is there any setting i miss?

the checkpoints are downloaded from google drive (checkpoints_all.tar.gz).

**Comment by BFeng14** [comment]
Thank you for your interest on Actfound, there are several things you can check:
1. please tell me the name of checkpoints you are using to check its correctness
2. using Actfound(transfer) for fusion: as reported in our paper, we use the first step loss to fusion Actfound and Actfound(transfer), please run Actfound(transfer) as well, and using fusion_result.py to get the final prediction results.
3. number of repeating experiments: the results of are reported as the average of 40 repeating experiments. I found that the result of 0.6 for finetuning is better than 0.8 as finetuning, which doesn't make sense as using more data for finetuning can give better results. 

Thank you again for your interest on Actfound, I will add a step-by-step tutorial for reproducing FEP results after this, and your help is greatly appreated.

**Comment by zfwangDP** [comment]
Thanks for replying.
The model is run using the provided code:

`FIXED_PARAM="--test_sup_num [0.2,0.4,0.6,0.8] --test_repeat_num 1 --train 0 --test_epoch -1 --expert_test fep"

CHEMBL_DIR="./checkpoints_all/checkpoints_chembl"
CHEMBL_RES="./test_results/result_fep_new/fep/chembl"
python main_reg.py --datasource=chembl --logdir ${CHEMBL_DIR}/checkpoint_chembl_actfound --model_name actfound --test_write_file ${CHEMBL_RES} ${FIXED_PARAM} &`

it seems that the parameters are loaded from "./checkpoints_all/checkpoints_chembl/checkpoint_chembl_actfound/checkpoint_chembl_actfound/model_best"

I do miss the result fusion process, i will try it later.
Number of repeating experiments may account for the lower r2 for 0.8 split as i just run a single test.
It will be pretty appreciated if a step by step tutorial is provided, thanks again.

**Comment by zfwangDP** [comment]
The results are consistent as i recover "--test_repeat_num" to 40. 
By the way, what's the differences between the repeat runs? Is the data randomly split every time?

**Comment by BFeng14** [comment]
yes, each repeat randomly split every time. "results are consistent" means that the average r2 is still 0.449,0.550,0.604 and 0.587 for sup_num 0.2,0.4,0.6,0.8 ?

**Comment by zfwangDP** [comment]
sorry for confusing abbreviation.. I mean that i successfully reproduce the reported results and the results are consistent with the files given in test_results_all.tar.gz.
I think my problem is solved. Thanks~
